### PR TITLE
feat: add grains exercise

### PR DESCRIPTION
**Pull request by vaeng**

### CODE REVIEW

File: exercises/practice/grains/create_test_table.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+INSERT INTO tests (uuid, name, task, square, expected)
+    VALUES
+        ("9fbde8de-36b2-49de-baf2-cd42d6f28405", "grains on square 1", "single-square", 1, 1),
+        ("ee1f30c2-01d8-4298-b25d-c677331b5e6d", "grains on square 2", "single-square", 2, 2),
+        ("10f45584-2fc3-4875-8ec6-666065d1163b", "grains on square 3", "single-square", 3, 4),
+        ("a7cbe01b-36f4-4601-b053-c5f6ae055170", "grains on square 4", "single-square", 4, 8),
+        ("c50acc89-8535-44e4-918f-b848ad2817d4", "grains on square 16", "single-square", 16, 32768),
+        ("acd81b46-c2ad-4951-b848-80d15ed5a04f", "grains on square 32", "single-square", 32, 2147483648),
+        ("c73b470a-5efb-4d53-9ac6-c5f6487f227b", "grains on square 64", "single-square", 64, 9223372036854775808),
+        ("6eb07385-3659-4b45-a6be-9dc474222750", "returns the total number of grains on the board", "total", "", 18446744073709551615);
```
Comment (IsaacG) [inline exercises/practice/grains/create_test_table.sql:29]: This sets the square to a string. The square is an int field.

File: exercises/practice/grains/.meta/tests.toml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+[acd81b46-c2ad-4951-b848-80d15ed5a04f]
+description = "returns the number of grains on the square -> grains on square 32"
+
+[c73b470a-5efb-4d53-9ac6-c5f6487f227b]
+description = "returns the number of grains on the square -> grains on square 64"
+
+[1d47d832-3e85-4974-9466-5bd35af484e3]
+description = "returns the number of grains on the square -> square 0 is invalid"
+include = false
+comment = "error testing omitted on purpose"
```
Comment (glennj) [inline exercises/practice/grains/.meta/tests.toml:36]: 👍

**Review by glennj — APPROVED** [review]

**Comment by glennj** [comment]
```
grains: testing...
/home/runner/work/sqlite/sqlite/exercises/practice/grains
Test failed for grains - failing tests:
{
  "name": "returns the total number of grains on the board",
  "message": null
}
```
Are we overflowing here? `SET result = POWER(2, 64) - 1`

**Comment by vaeng** [comment]
> Are we overflowing here? 

```bash
❯ sqlite3
SQLite version 3.37.2 2022-01-06 13:25:41
sqlite> select POWER(2, 64) - 1;
1.84467440737096e+19
sqlite> .quit
❯ python3
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
>>> 2**64 -1
18446744073709551615
```

@glennj seems okay, I can't find any hard limits.

**Pull request merged**